doc/source/data/glossary.rst

```diff
@@ -0,0 +1,163 @@
+.. _datasets_glossary:
+
+=====================
+Ray Datasets Glossary
+=====================
+
+.. glossary::
+
+    Batch format
+        The way batches of data are represented.
+
+        Set ``batch_format`` in methods like
+        :meth:`Dataset.iter_batches() <ray.data.Dataset.iter_batches>` and
+        :meth:`Dataset.map_batches() <ray.data.Dataset.map_batches>` to specify the
+        batch type.
+
+        .. doctest::
+
+            >>> import ray
+            >>> dataset = ray.data.range_table(10)
+            >>> next(iter(dataset.iter_batches(batch_format="numpy", batch_size=5)))
+            {'value': array([0, 1, 2, 3, 4])}
+            >>> next(iter(dataset.iter_batches(batch_format="pandas", batch_size=5)))
+               value
+            0      0
+            1      1
+            2      2
+            3      3
+            4      4
+
+        To learn more about batch formats, read
+        :ref:`UDF Input Batch Formats <transform_datasets_batch_formats>`.
+
+    Block
+        A processing unit of data. A :class:`~ray.data.Dataset` consists of a
+        collection of blocks.
+
+        Under the hood, :term:`Datasets <Datasets (library)>` partition :term:`records <Record>`
+        into a set of distributed data blocks. This allows Datasets to perform operations
+        in parallel.
+
+        Unlike a batch, which is a user-facing object, a block is an internal abstraction.
+
+    Block format
+        The way :term:`blocks <Block>` are represented.
+
+        Blocks are represented as
+        `Arrow tables <https://arrow.apache.org/docs/python/generated/pyarrow.Table.html>`_,
+        `pandas DataFrames <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_,
+        and Python lists. To determine the block format, call
+        :meth:`Dataset.dataset_format() <ray.data.Dataset.dataset_format>`.
+
+    Data Shuffling
```

Comment:
Should we talk about Datasets operations (transformation and consumption) instead (or first)?
Shuffle is just one of the transforms, and `random_shuffle` (global shuffle) isn't really used widely.

Reply:
Could you elaborate on what you have in mind? Like, are you suggesting we add glossary terms for "Dataset transformation" and "Dataset consumption"?

Reply:
yeah, we can also combine them into "Dataset operations" and explain that it can be transforms and consumptions (like Spark also does this https://spark.apache.org/docs/latest/rdd-programming-guide.html#rdd-operations).

Reply:
To make progress here, I think we may drop shuffling and then make a followup PR to fix. This is not a blocker.

Reply:
Created issue to track https://github.com/ray-project/ray/issues/32556